### ISSUE TITLE
Support for circular dependencies - eg have `model` as both input and ouput of `train` stage

### ISSUE THREAD

**Issue by davesteps** [iterative/dvc#4724]
It quite common when training deep learning models not to start training from scratch every time but rather use a previous model as initial starting weights. For example you may train an basic model then make some adjustments to your training data or your loss function or adjust some other parameter and continue training.  

I am currently using a solution suggested on the forum of using `import` to import a previous version of my model. However, it was also suggested that a better solution could be supported where the `model` object could be both an input and output of your `train` stage.

The original question for more context:
https://discuss.dvc.org/t/use-previous-version-of-model-as-initial-weights/525/2

**Comment by karajan1001** [comment]
I have the same problem. Currently, I didn't put `model` into the dependency of the `train` stage and just maintain it manually.

**Comment by efiop** [comment]
@davesteps @karajan1001 We have an upcoming experimental feature for experiment checkpoints https://github.com/iterative/dvc/issues/4498 , maybe it would be useful in your scenario as well?

**Comment by davesteps** [comment]
@efiop It sounds like the checkpoints feature is very close to what I would like to achieve. I'm certainly willing to give it a try when it is available. I also wasn't aware of the `--outs-persist` option which could be useful. Thanks.

**Comment by efiop** [comment]
@davesteps There is no official documentation yet and it is still an experimental feature, but we have a wiki for it https://github.com/iterative/dvc/wiki/Experiments#checkpoint-experiments . Consider taking a look.

**Comment by efiop** [comment]
@davesteps Checkpoints have been released in 2.0, did you have a chance to try them?

**Comment by efiop** [comment]
Closing as stale for now.
